nemo/collections/asr/parts/submodules/rnnt_beam_decoding.py

```diff
@@ -325,6 +328,9 @@ def __init__(
         else:
             self.ngram_lm = None
 
+        self.hat_subtract_ilm = hat_subtract_ilm
+        self.hat_ilm_weight = hat_ilm_weight
+
```

Comment:
Inside the  `__call__`, see if the ilm flag is set in the self.hat_subtract_ilm (self is the decoding beam search class), do a hasattr check again and now set the value. 

Keep a coph of the original value and then at the end of the call function, before returning, set the ilm value back to the original value. 

All of this is so that 1) hat model is by default running like a normal RNNT model. 2) only inside beam search it will become used as the ilm result.